packages/form/src/components/KeyValueField/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { Button, Row, Stack } from '@ultraviolet/ui'
+import type { ComponentProps } from 'react'
+import { useFieldArray } from 'react-hook-form'
+import { TextInputField } from '../TextInputField'
+
+type InputProps = {
+  label: ComponentProps<typeof TextInputField>['label']
+  required?: ComponentProps<typeof TextInputField>['required']
+  regex?: ComponentProps<typeof TextInputField>['regex']
+}
+
+type InputValueProps = {
+  type?: ComponentProps<typeof TextInputField>['type']
+  placeholder?: ComponentProps<typeof TextInputField>['placeholder']
+} & InputProps
+
+type AddButtonProps = {
+  name: ComponentProps<typeof Button>['children']
+  fullWidth?: ComponentProps<typeof Button>['fullWidth']
+  tooltip: string
+  tooltipAlert: string
+}
+
+type KeyValueFieldProps = {
+  name: string
+  inputKey: InputProps
+  inputValue: InputValueProps
+  addButton: AddButtonProps
+  maxSize?: number
+  readonly?: boolean
+}
+
+export type KeyValue = { key: string; value: string }
+
+export const KeyValueField = ({
```

Comment:
We need to provide the form values as a generic or the react-hook-form `control`.